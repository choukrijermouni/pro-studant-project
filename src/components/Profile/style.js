import {
  neutral,
  blue
} from '@pro_boa/ui'
export default {
  root: {
    display: 'flex',
    alignItems: 'center'
  },
  avatar: {
    margin: '16px 0'
  },
  iconHeader: {
    fontSize: 22,
    float: 'right',
    marginRight: 12,
    marginTop: 24,
    marginBottom: 24,
    color: neutral[4],
    cursor: 'pointer'
  },
  container: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    padding: '0 40px'
  },
  notificationStyle: {
    boxSizing: 'border-box',
    height: 21,
    width: 21,
    border: '1px solid #F8F8F8',
    borderRadius: 10,
    backgroundColor: 'rgba(233,27,75,0.99)',
    boxShadow: '0 9px 14px -10px rgb(0 0 0 / 50%)',
    position: 'absolute',
    top: 25.5,
    right: 293
  },
  alert: {
    height: 13,
    width: 20,
    color: '#FFFFFF',
    fontFamily: 'Muli',
    fontSize: 10,
    fontWeight: 'bold',
    letterSpacing: 1.23,
    lineHeight: 2,
    textAlign: 'center'
  },
  empty: {
    display: 'none'
  },
  notificationPaper: {
    width: 449,
    height: 603,
    backgroundColor: neutral[0],
    position: 'relative',
    top: 24,
    right: 250,
    display: 'flex',
    flexDirection: 'column'
  },
  notificationCardsContainer: {
    width: ' 100%',
    height: 'calc(100% - 124px)',
    display: 'flex',
    flexDirection: 'column',
    overflow: 'scroll',
    overflowX: 'hidden',
    '&::-webkit-scrollbar': {
      width: 5
    },
    '&::-webkit-scrollbar-thumb': {
      background: blue[1],
      borderRadius: 50
    }
  },
  notificationContainer: {
    width: 20,
    height: 20
  },
  notificationHeader: {
    height: 78,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    width: '100%',
    padding: 12
  },
  notificationCard: {
    height: 95,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    width: '100%',
    borderBottom: `.3px ${neutral[2]} solid`,
    padding: 12
  },
  notificationFooter: {
    height: 47,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    width: '100%',
    position: 'absolute',
    backgroundColor: neutral[0],
    bottom: 0,
    padding: 12,
    borderTop: `.3px ${neutral[2]} solid`
  },
  avatarContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-start',
    width: '90%'
  },
  icon: {
    color: blue[0],
    fontWeight: 200
  },
  textContainer: {
    marginLeft: 16
  },
  triangle: {
    position: 'relative',
    top: '-17px',
    left: 213
  },
  avatarPopoverContainer: {
    display: 'flex',
    padding: 16,
    alignItems: 'center',
    justifyContent: 'flex-start'
  },
  profileInfoContainer: {
    marginLeft: 8
  },
  devider: {
    width: '90%',
    height: 1,
    backgroundColor: neutral[2]
  },
  deviderContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    width: '100%'
  },
  bigDeviderContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    width: '100%'
  },
  linkContainer: {
    padding: 16,
    width: '100%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    transition: 'background 0.3s ease',
    '& > p': {
      transition: 'color 0.3s ease',
      color: neutral[5]
    },
    '&:hover': {
      backgroundColor: neutral[1]
    },
    '&:hover > p': {
      color: neutral[6]
    }
  },
  iconStyle: {
    height: 12,
    width: 12,
    color: blue[0],
    lineHeight: 0,
    marginLeft: 8
  },
  contactContainer: {
    cursor: 'pointer',
    width: '100%',
    padding: 16,
    display: 'flex',
    color: neutral[5],
    alignItems: 'center',
    justifyContent: 'flex-start',
    transition: 'background 0.3s ease',
    '&:hover': {
      color: neutral[6],
      backgroundColor: neutral[1]
    }
  },
  contactIconStyle: {
    fontSize: 18,
    color: blue[0],
    lineHeight: 0,
    marginRight: 8
  },
  linkTextStyle: {
    transition: 'background 0.3s ease, color 0.3s ease',
    '&:hover': {
      backgroundColor: neutral[1]
    }
  },
  linkStyle: {
    textDecoration: 'none'
  }
}
