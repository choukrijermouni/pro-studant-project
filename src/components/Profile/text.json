{
    "profile": "Mon compte",
    "logout": "Se déconnecter",
    "links": {
        "start": "Bien commencer",
        "admin": "Planifier votre réussite",
        "understand": "Comprendre les rapports",
        "news": "Nouveautés pro_boa",
        "manager": "Guide du manager"
    },
    "qanda": "Q&A",
    "help": "Centre d'aide",
    "contact": "Contactez-nous",
    "roleLabels": {
        "4": "Administrateur",
        "5": "Manager"
    }
}