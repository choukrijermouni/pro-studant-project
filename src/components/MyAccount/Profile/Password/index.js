import {
  PasswordInput,
  TypographyElement,
  neutral,
  Button
} from '@pro_boa/ui'
import Text from './text.json'
import style from './style'
import { createUseStyles } from 'react-jss'
import { useFormik } from 'formik'
import { editOrganizationProfilePasswordAction } from 'pages/MyAccount/store'
import { useDispatch } from 'react-redux'

const useStyle = createUseStyles(style)

export default () => {
  const dispatch = useDispatch()
  const { formClass, container } = useStyle()
  const { handleChange: onFormikHandleChange, handleSubmit, values } = useFormik({
    initialValues: {
      password: ''
    },
    onSubmit: (values) => {
      dispatch(editOrganizationProfilePasswordAction(values.password))
      window.scrollTo(0, 0)
    }
  })
  return (
    <div className={container}>
      <TypographyElement
        variant='heading2'
        color={neutral[6]}
        fontSize='16px'
        spacing='24px 0 0 0'
        lineHeight='20px'
      >
        {Text.placeHolder}
      </TypographyElement>
      <form onSubmit={handleSubmit} className={formClass}>
        <PasswordInput
          dataTest='password-input'
          handleChange={onFormikHandleChange}
          value={values.recommendation}
          type='password'
          label={Text.title}
          id='password'
          placeholder={Text.title}
        />
        <Button
          dataTest='password-submit-button'
          handleClick={handleSubmit}
          type='submit'
          label={Text.buttonLabel}
          width='200px'
          disabled={!values.password}
        />
      </form>
    </div>
  )
}
