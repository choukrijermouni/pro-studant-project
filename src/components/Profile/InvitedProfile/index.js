import GoBackAction from 'components/Common/GoBackAction'
import Layout from 'components/Common/Layout'
import { useDispatch, useSelector } from 'react-redux'
import {
  Avatar,
  TypographyElement,
  neutral,
  AvatarSize,
  red,
  Icon,
  Icons
} from '@pro_boa/ui'
import { amazonBucket, Avatars, teams } from 'constants/'
import { createUseStyles } from 'react-jss'
import style from './style'
import Text from './text.json'
import { cancelLearnerInvitationAction } from 'pages/LearnerProfile/store'
import DeleteInvitationModal from 'components/Common/InviteBar/DeleteInvitationModal'
import { useEffect, useState } from 'react'
import { goBack, push } from 'connected-react-router'
import { TeamDetailsNeutralPath } from 'Routes'
import classNames from 'classnames'

const useStyles = createUseStyles(style)

const defaultTeamImageIndex = 0

export default () => {
  const dispatch = useDispatch()
  const { InvitationId, InvitationEmail, TeamId, TeamImage, TeamName, InvitationLicenseType } = useSelector(({ learners }) => learners?.invitedLearner)
  const { organizationLicenseTypes = {} } = useSelector(({ referential }) => referential)
  const { header, actionContainer, teamIcon, icon, card, cursor } = useStyles()
  const [openDeleteInvitationModal, setOpenDeleteInvitationModal] = useState(false)
  useEffect(() => {
    if (!InvitationId || !InvitationEmail) {
      dispatch(goBack())
    }
  }, [InvitationId, InvitationEmail])
  return (
    <Layout>
      <GoBackAction />
      <div className={header}>
        <Avatar
          size={AvatarSize.size_4}
          img={`${amazonBucket.avatar}${Avatars.anonymous}`}
          dot
          dotColor={red[3]}
          dotWidth={20}
          dotHeight={20}
          dotTop={8}
          dotRight={8}
        />
        <div>
          <TypographyElement
            component='h3'
            variant='heading3'
            align='left'
            fontSize='30px'
            spacing='20px 10px 30px 20px'
            color={neutral[6]}
          >
            {InvitationEmail}
          </TypographyElement>
          <div className={header}>
            <div
              className={classNames(card, cursor)}
              onClick={() => TeamId && dispatch(push(`${TeamDetailsNeutralPath}/${TeamId}`))}
            >
              <img src={TeamImage || teams[defaultTeamImageIndex]} alt='team' className={teamIcon} />
              <TypographyElement
                component='h4'
                variant='caption1'
                align='left'
                spacing='10px 16px 11px 16px'
                color={neutral[5]}
              >
                {TeamName || Text.withoutTeam}
              </TypographyElement>
              <Icon iconName={Icons.teams} style={icon} />
            </div>
            <div className={card}>
              <Avatar name={InvitationLicenseType ? organizationLicenseTypes[InvitationLicenseType]?.charAt(0) : 'S'} size={AvatarSize.size_0} />
              <TypographyElement
                component='h4'
                variant='caption1'
                align='left'
                spacing='10px 16px 11px 16px'
                color={neutral[5]}
              >
                {InvitationLicenseType ? `${Text.license} ${Text.licenseTypes[organizationLicenseTypes[InvitationLicenseType]]}` : Text.withoutLicense}
              </TypographyElement>
            </div>
          </div>
        </div>
      </div>
      <div className={actionContainer}>
        <TypographyElement
          color={red[3]}
          align='center'
          spacing='0px 20px 0 20px'
          display='inline'
          cursor='pointer'
          handleClick={() => setOpenDeleteInvitationModal(true)}
        >
          {Text.cancelInvitation}
        </TypographyElement>
      </div>
      <DeleteInvitationModal
        openModal={openDeleteInvitationModal}
        handleClose={() => setOpenDeleteInvitationModal(false)}
        handleSubmit={() => {
          dispatch(cancelLearnerInvitationAction(InvitationId))
          setOpenDeleteInvitationModal(false)
        }}
      />
    </Layout>
  )
}
