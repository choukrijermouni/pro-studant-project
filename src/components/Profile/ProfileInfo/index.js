import { createUseStyles } from 'react-jss'
import { useEffect, useState } from 'react'
import {
  Avatar,
  AvatarSize,
  TypographyElement,
  neutral,
  Icon,
  red,
  Icons
} from '@pro_boa/ui'
import style from './style'
import Text from './text.json'
import { push } from 'connected-react-router'
import { useDispatch, useSelector } from 'react-redux'
import { amazonBucket, DMYDateFormat, teams } from 'constants/'
import {
  TeamDetailsNeutralPath,
  AffectationHistoryPath
} from 'Routes'
import moment from 'moment'
import { avatarName, isExpired } from 'helpers'
import { useParams } from 'react-router-dom'
import { Popover, PopoverContent, PopoverTrigger } from 'components/Common/PopOver'
import { requestCoursesAccessAction, resendWelcomeMessageAction } from 'pages/Learners/store'
import { useDrawer } from 'components/Common/Drawer/drawerContext'
import {
  fetchLearnerProgressionAction,
  sendNotificationAction,
  switchLastLicensePopupAction
} from 'pages/LearnerProfile/store'
import LastLicenseModal from 'components/Common/Drawer/AffectLicense/LastLicenseModal'

const defaultTeamImageIndex = 0

const useStyles = createUseStyles(style)
const noDate = true
export default ({
  size,
  manager,
  Name,
  Email,
  LastConnectionDate,
  EndDate,
  IsActive,
  Photo,
  TeamId,
  TeamImage,
  TeamName
}) => {
  const dispatch = useDispatch()
  const { id } = useParams()
  const { openDrawer, closeDrawer } = useDrawer()
  const {
    header,
    teamIcon,
    icon,
    card,
    iconLastLogin,
    infos,
    cardWithoutPointer,
    dotsPopoverContainer,
    popOverContent,
    linkTextStyle,
    disabledLinkTextStyle,
    dot,
    linkContainer,
    disabledLinkContainer
  } = useStyles()
  const { TotalLicensesRemaining } = useSelector(({ organization }) => organization.licenseInfo)
  const { OrganizationLicenseType } = useSelector(state => state.profile)
  const { organizationLicenseTypes } = useSelector(({ referential }) => referential)
  const [isOpen, setIsOpen] = useState(false)
  const { emailTypes } = useSelector(({ referential }) => referential)
  useEffect(() => {
    dispatch(fetchLearnerProgressionAction(id))
  }, [id])
  const { lastLicensePopupClosed } = useSelector(({ profile }) => profile)
  useEffect(() => {
    setIsOpen(!lastLicensePopupClosed)
  }, [lastLicensePopupClosed])
  const scale = window.devicePixelRatio
  const popOverContentMap = [
    {
      text: Text.affectLicense,
      onHandleClick: () => TotalLicensesRemaining
        ? openDrawer({
            componentName: 'affectLicense',
            props: {
              id,
              handleClose: closeDrawer
            }
          })
        : null,
      className: TotalLicensesRemaining ? linkTextStyle : disabledLinkTextStyle,
      containerClassName: TotalLicensesRemaining ? linkContainer : disabledLinkContainer,
      color: TotalLicensesRemaining ? neutral[6] : neutral[5]
    },
    {
      text: Text.reinviteLearner,
      onHandleClick: () => LastConnectionDate === null
        ? dispatch(resendWelcomeMessageAction([id], noDate))
        : null,
      className: LastConnectionDate === null ? linkTextStyle : disabledLinkTextStyle,
      containerClassName: LastConnectionDate === null ? linkContainer : disabledLinkContainer,
      color: LastConnectionDate === null ? neutral[6] : neutral[5]
    },
    {
      text: Text.editTeam,
      onHandleClick: () => openDrawer(
        {
          componentName: 'editLearnerTeam',
          props: {
            id: TeamId,
            closeModal: closeDrawer
          }
        }
      )
    },
    {
      text: Text.deleteLearner,
      onHandleClick: () => openDrawer(
        {
          componentName: 'removeLearner',
          props: {
            isLearnerProfile: true,
            handleClose: closeDrawer
          }
        }
      ),
      color: red[3]
    }
  ]
  const managerPopOverContentMap = [
    {
      text: Text.reinviteLearner,
      onHandleClick: () => LastConnectionDate === null ? dispatch(resendWelcomeMessageAction([id], noDate)) : null,
      className: LastConnectionDate === null ? linkTextStyle : disabledLinkTextStyle,
      containerClassName: LastConnectionDate === null ? linkContainer : disabledLinkContainer,
      color: LastConnectionDate === null ? neutral[6] : neutral[5]
    },
    {
      text: Text.askAccess,
      onHandleClick: () => dispatch(requestCoursesAccessAction(id))
    }
  ]
  const formattedPopOverContentMap = LastConnectionDate !== null ? popOverContentMap.filter(e => e.text !== Text.reinviteLearner) : popOverContentMap
  const formattedManagerPopOverContentMap = LastConnectionDate !== null ? managerPopOverContentMap.filter(e => e.text !== Text.reinviteLearner) : managerPopOverContentMap
  return (
    <>
      <LastLicenseModal
        openModal={isOpen}
        handleClose={() => {
          dispatch(sendNotificationAction(emailTypes.OrganizationLicensesDepleted))
          dispatch(switchLastLicensePopupAction(true))
          setIsOpen(false)
        }}
        handleSubmit={() => {
          dispatch(sendNotificationAction(emailTypes.OrganizationLicensesRequested))
          dispatch(switchLastLicensePopupAction(true))
          setIsOpen(false)
        }}
      />
      <div className={infos}>
        <Avatar
          size={size}
          img={Photo ? `${amazonBucket.bucketBaseUrl}${Photo}` : null}
          name={avatarName(Name)}
          IsActive={IsActive}
        />
        <div>
          <div className={infos}>
            <TypographyElement
              component='h3'
              variant='heading3'
              align='left'
              fontSize={scale > 1 ? '20px' : '30px'}
              spacing='20px 10px 30px 20px'
              color={neutral[6]}
            >
              {Name || ''}
            </TypographyElement>
            <TypographyElement
              component='h4'
              variant='caption1'
              align='left'
              spacing='23px 0 30px 0'
              color={neutral[5]}
            >
              {Email}
            </TypographyElement>
            <Popover>
              <PopoverTrigger>
                <div className={dotsPopoverContainer}>
                  <span className={dot} />
                  <span className={dot} />
                  <span className={dot} />
                </div>
              </PopoverTrigger>
              <PopoverContent topPosition={40}>
                {
                  !manager
                    ? (
                      <div className={popOverContent}>
                        {formattedPopOverContentMap.map((item, index) => (
                          <div
                            key={index}
                            className={item.containerClassName || linkContainer}
                            onClick={() => {
                              item.onHandleClick()
                            }}
                          >
                            <TypographyElement
                              key={index}
                              color={item.color || neutral[6]}
                              fontWeight='normal'
                              fontSize='13px'
                              lineHeight='16px'
                              className={item.className || linkTextStyle}
                            >
                              {item.text}
                            </TypographyElement>
                          </div>
                        ))}
                      </div>)
                    : (
                      <div className={popOverContent}>
                        {formattedManagerPopOverContentMap.map((item, index) => (
                          <div
                            key={index}
                            className={linkContainer}
                            onClick={() => {
                              item.onHandleClick()
                            }}
                          >
                            <TypographyElement
                              key={index}
                              handleClick={item.onHandleClick}
                              color={item.color || neutral[6]}
                              fontWeight='normal'
                              fontSize='13px'
                              lineHeight='16px'
                              className={item.className || linkTextStyle}
                            >
                              {item.text}
                            </TypographyElement>
                          </div>
                        ))}
                      </div>)
                }
              </PopoverContent>
            </Popover>
          </div>
          <div className={header}>
            {
              !manager && (
                <>
                  <div
                    data-test='learner-team-name'
                    className={card}
                    onClick={() => TeamId && dispatch(push(`${TeamDetailsNeutralPath}/${TeamId}`))}
                  >
                    {TeamId
                      ? (
                        <img
                          src={TeamImage || teams[defaultTeamImageIndex]}
                          alt='team' className={teamIcon}
                        />)
                      : null}
                    <TypographyElement
                      component='h4'
                      variant='caption1'
                      align='left'
                      spacing='10px 16px 11px 16px'
                      color={neutral[5]}
                    >
                      {TeamName || Text.withoutTeam}
                    </TypographyElement>
                    <Icon iconName={Icons.teams} style={icon} />
                  </div>
                  <div
                    className={card}
                    onClick={() => dispatch(push(`${AffectationHistoryPath}/${id}`))}
                  >
                    <Avatar
                      name={EndDate && moment(EndDate).isAfter(moment())
                        ? organizationLicenseTypes[OrganizationLicenseType]?.charAt(0)
                        : 'S'} size={AvatarSize.size_0}
                    />
                    <TypographyElement
                      dataTest='license-assignment-history'
                      component='h4'
                      variant='caption1'
                      align='left'
                      spacing='10px 16px 11px 16px'
                      color={neutral[5]}
                    >
                      {
                        EndDate
                          ? isExpired(EndDate)
                              ? Text.withoutLicense
                              : `${Text.license} ${Text.licenseTypes[organizationLicenseTypes[OrganizationLicenseType]]} ${Text.expire} ${moment(EndDate).format(DMYDateFormat)}`
                          : Text.withoutLicense
                      }
                    </TypographyElement>
                  </div>
                </>)
            }
            <div className={cardWithoutPointer}>
              <Icon iconName={Icons.lastUserLogin} style={iconLastLogin} />
              <TypographyElement
                component='h4'
                variant='caption1'
                align='left'
                spacing='10px 16px 11px 16px'
                color={neutral[5]}
              >
                {LastConnectionDate !== null
                  ? `${Text.lastConnection} ${moment(LastConnectionDate).format(DMYDateFormat)}`
                  : Text.neverConnected}
              </TypographyElement>
            </div>
          </div>
        </div>
      </div>
    </>
  )
}
