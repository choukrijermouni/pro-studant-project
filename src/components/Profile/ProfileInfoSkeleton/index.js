import { createUseStyles } from 'react-jss'
import {
  Avatar,
  Skeleton
} from '@pro_boa/ui'
import style from './style'

const useStyles = createUseStyles(style)
export default ({ manager }) => {
  const {
    header,
    infos,
    skeletonContainer,
    titleStyle
  } = useStyles()
  return (
    <>
      <div className={infos}>
        <Avatar
          size='size_4'
          loading
        />
        <div>
          <div className={titleStyle}>
            <Skeleton lines={1} height={40} margin='20px 10px 30px 20px' width='100%' />
            <Skeleton lines={1} height={40} margin='20px 0 30px' width={50} />
          </div>
          <div className={header}>
            {
              !manager &&
              (
                <>
                  <div className={skeletonContainer}><Skeleton lines={1} height={40} margin='0px 0px 0px 20px' width={170} /></div>
                  <div className={skeletonContainer}><Skeleton lines={1} height={40} margin='0px 0px 0px 20px' width={170} /></div>
                </>)
            }
            <div className={skeletonContainer}><Skeleton lines={1} height={40} margin='0px 0px 0px 20px' width={170} /></div>
          </div>
        </div>
      </div>
    </>
  )
}
