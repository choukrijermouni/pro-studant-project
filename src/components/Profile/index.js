import {
  Avatar,
  TypographyElement,
  neutral,
  red,
  Icon,
  Icons,
  blue
} from '@pro_boa/ui'
import { createUseStyles } from 'react-jss'
import style from './style'
import { externalLink, help } from 'assets'
import { useDispatch, useSelector } from 'react-redux'
import { push } from 'connected-react-router'
import { MyAccountPath } from 'Routes'
import { helpLinks, amazonBucket } from 'constants/'
import { Popover, PopoverContent, PopoverTrigger } from 'components/Common/PopOver'
import Text from './text'
import { tryLogout } from 'identity/store'
import { avatarName } from 'helpers'

const useStyle = createUseStyles(style)

export default () => {
  const { avatar, iconHeader, container, root, avatarPopoverContainer, linkStyle, profileInfoContainer, devider, deviderContainer, linkContainer, iconStyle, bigDeviderContainer, contactContainer, contactIconStyle, linkTextStyle } = useStyle()
  const { Photo } = useSelector(state => state.organizationProfile)
  const { FirstName, LastName } = useSelector(state => state.organizationProfile)
  const { user } = useSelector(state => state.identity)
  const dispatch = useDispatch()
  const name = `${FirstName} ${LastName}`
  const { isManager } = useSelector(({ organizationProfile }) => organizationProfile)
  return (
    <div className={root}>
      <div>
        <div className={container}>
          <Popover>
            <PopoverTrigger>
              <a target='_blank' rel='noreferrer'>
                <img src={help} alt='notification' className={iconHeader} />
              </a>
            </PopoverTrigger>
            <PopoverContent>
              <div>
                <a href={helpLinks.start} className={linkStyle} target='_blank' rel='noreferrer'>
                  <div className={linkContainer}>
                    <TypographyElement
                      component='p'
                      color={neutral[6]}
                      fontWeight='bold'
                      fontSize='16px'
                      lineHeight='20px'
                    >
                      {Text.links.start}
                    </TypographyElement>
                    <img src={externalLink} alt='external' className={iconStyle} />
                  </div>
                </a>
                {isManager
                  ? (
                    <a href={helpLinks.managerGuid} className={linkStyle} target='_blank' rel='noreferrer'>
                      <div className={linkContainer}>
                        <TypographyElement
                          component='p'
                          color={neutral[6]}
                          fontWeight='bold'
                          fontSize='16px'
                          lineHeight='20px'
                        >
                          {Text.links.manager}
                        </TypographyElement>
                        <img src={externalLink} alt='external' className={iconStyle} />
                      </div>
                    </a>)
                  : (
                    <a href={helpLinks.admins} className={linkStyle} target='_blank' rel='noreferrer'>
                      <div className={linkContainer}>
                        <TypographyElement
                          component='p'
                          color={neutral[6]}
                          fontWeight='bold'
                          fontSize='16px'
                          lineHeight='20px'
                        >
                          {Text.links.admin}
                        </TypographyElement>
                        <img src={externalLink} alt='external' className={iconStyle} />
                      </div>
                    </a>)}
                <a href={helpLinks.undestand} className={linkStyle} target='_blank' rel='noreferrer'>
                  <div className={linkContainer}>
                    <TypographyElement
                      component='p'
                      color={neutral[6]}
                      fontWeight='bold'
                      fontSize='16px'
                      lineHeight='20px'
                    >
                      {Text.links.understand}
                    </TypographyElement>
                    <img src={externalLink} alt='external' className={iconStyle} />
                  </div>
                </a>
                <a href={helpLinks.news} className={linkStyle} target='_blank' rel='noreferrer'>
                  <div className={linkContainer}>
                    <TypographyElement
                      component='p'
                      color={neutral[6]}
                      fontWeight='bold'
                      fontSize='16px'
                      lineHeight='20px'
                    >
                      {Text.links.news}
                    </TypographyElement>
                    <img src={externalLink} alt='external' className={iconStyle} />
                  </div>
                </a>
                <div className={bigDeviderContainer}>
                  <div className={devider} />
                </div>

                <a href={helpLinks.faq} className={linkStyle} target='_blank' rel='noreferrer'>
                  <div className={linkContainer}>
                    <TypographyElement
                      component='p'
                      color={neutral[6]}
                      fontWeight='bold'
                      fontSize='16px'
                      lineHeight='20px'
                    >
                      {Text.qanda}
                    </TypographyElement>
                    <img src={externalLink} alt='notification' className={iconStyle} />
                  </div>
                </a>

                <a href={helpLinks.help} className={linkStyle} target='_blank' rel='noreferrer'>
                  <div className={linkContainer}>
                    <TypographyElement
                      component='p'
                      color={neutral[6]}
                      fontWeight='bold'
                      fontSize='16px'
                      lineHeight='20px'
                    >
                      {Text.help}
                    </TypographyElement>
                    <img src={externalLink} alt='notification' className={iconStyle} />
                  </div>
                </a>
                <div className={contactContainer} onClick={() => window.Intercom('showNewMessage')}>
                  <Icon iconName={Icons.support} style={contactIconStyle} />
                  <TypographyElement
                    color={blue[0]}
                    fontWeight='bold'
                    fontSize='16px'
                    lineHeight='20px'
                  >
                    {Text.contact}
                  </TypographyElement>
                </div>
              </div>
            </PopoverContent>
          </Popover>

          <Popover>
            <PopoverTrigger>
              <div className={avatar}>
                <Avatar img={Photo ? `${amazonBucket.bucketBaseUrl}${Photo}` : null} name={avatarName(name)} size='size_1' />
              </div>
            </PopoverTrigger>
            <PopoverContent>
              <div>
                <div className={avatarPopoverContainer}>
                  <Avatar img={Photo ? `${amazonBucket.bucketBaseUrl}${Photo}` : null} name={avatarName(name)} size='size_1' />
                  <div className={profileInfoContainer}>
                    <TypographyElement
                      color={neutral[6]}
                      fontWeight='bold'
                      fontSize='16px'
                      lineHeight='20px'
                    >
                      {FirstName} {LastName}
                    </TypographyElement>
                    <TypographyElement
                      color={neutral[5]}
                      fontWeight='regular'
                      fontSize='14px'
                      lineHeight='18px'
                    >
                      {Text.roleLabels[user?.role]}
                    </TypographyElement>
                  </div>
                </div>
                <TypographyElement
                  className={linkTextStyle}
                  fontWeight='bold'
                  fontSize='16px'
                  margin='16px 0 0 0'
                  spacing='16px'
                  lineHeight='20px'
                  cursor='pointer'
                  handleClick={() => dispatch(push(MyAccountPath))}
                >
                  {Text.profile}
                </TypographyElement>
                <div className={deviderContainer}>
                  <div className={devider} />
                </div>
                <TypographyElement
                  className={linkTextStyle}
                  color={red[4]}
                  fontWeight='bold'
                  fontSize='16px'
                  spacing='16px'
                  lineHeight='20px'
                  handleClick={() => dispatch(tryLogout())}
                  cursor='pointer'
                >
                  {Text.logout}
                </TypographyElement>
              </div>
            </PopoverContent>
          </Popover>
        </div>
      </div>
    </div>
  )
}
