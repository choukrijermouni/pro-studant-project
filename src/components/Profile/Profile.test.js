import { shallow, configure } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import renderer from 'react-test-renderer'
import Profile from './'
import { ReduxProvider } from '../../../test/ReduxProvider'

configure({ adapter: new Adapter() })

describe('>>> Profile --- Shallow Render Profile component', () => {
  let wrapper
  let tree

  beforeEach(() => {
    wrapper = shallow(<ReduxProvider><Profile /></ReduxProvider>)
    tree = renderer.create(
      <ReduxProvider>
        <Profile />
      </ReduxProvider>
    ).toJSON()
  })

  it('+++ render correctly and match snapshot', () => {
    expect(tree).toMatchSnapshot()
  })

  it('+++ render the component', () => {
    expect(wrapper).toBeTruthy()
  })
})
