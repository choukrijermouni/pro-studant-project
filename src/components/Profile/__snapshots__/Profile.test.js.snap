// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> Profile --- Shallow Render Profile component +++ render correctly and match snapshot 1`] = `
<div
  className="root-0-2-1"
>
  <div
    className="notificationContainer-0-2-11"
  >
    <div
      className="notificationStyle-0-2-6"
    >
      <div
        className="alert-0-2-7"
      >
        3
      </div>
    </div>
    <img
      alt="notification"
      className="iconHeader-0-2-4"
      onClick={[Function]}
      src={Object {}}
    />
  </div>
  <div>
    <div
      className="container-0-2-5"
    >
      <div
        className="avatar-0-2-2"
      >
        <span
          className="avatarGroup-0-2-19 avatarGroup-d0-0-2-20"
        >
          MD
        </span>
      </div>
      <div
        className="font-0-2-3"
      >
        <p
          className="text-0-2-21 text-d0-0-2-22"
          onClick={[Function]}
        >
          Mon Compte
        </p>
      </div>
    </div>
  </div>
  <div>
    <img
      alt="notification"
      className="iconHeader-0-2-4"
      src={Object {}}
    />
  </div>
</div>
`;
